From 811a9b21c1bb226236d2a880f9cbbcbcf6028c2e Mon Sep 17 00:00:00 2001
From: Matthias Clasen <mclasen@redhat.com>
Date: Fri, 12 Feb 2016 20:35:26 -0500
Subject: [PATCH] file chooser: Use the right model when getting the selection

We don't need to make assumptions about which model is currently
used since gtk_tree_selection_get_selection hands us the model.

https://bugzilla.gnome.org/show_bug.cgi?id=761757
---
 gtk/gtkfilechooserwidget.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/gtk/gtkfilechooserwidget.c b/gtk/gtkfilechooserwidget.c
index c77a8b7..6ef25b6 100644
--- a/gtk/gtkfilechooserwidget.c
+++ b/gtk/gtkfilechooserwidget.c
@@ -6266,10 +6266,11 @@ get_selected_file_info_from_file_list (GtkFileChooserWidget *impl,
   GtkTreeSelection *selection;
   GtkTreeIter iter;
   GFileInfo *info;
+  GtkTreeModel *model;
 
   g_assert (!priv->select_multiple);
   selection = gtk_tree_view_get_selection (GTK_TREE_VIEW (priv->browse_files_tree_view));
-  if (!gtk_tree_selection_get_selected (selection, NULL, &iter))
+  if (!gtk_tree_selection_get_selected (selection, &model, &iter))
     {
       *had_selection = FALSE;
       return NULL;
@@ -6277,7 +6278,7 @@ get_selected_file_info_from_file_list (GtkFileChooserWidget *impl,
 
   *had_selection = TRUE;
 
-  info = _gtk_file_system_model_get_info (priv->browse_files_model, &iter);
+  info = _gtk_file_system_model_get_info (GTK_FILE_SYSTEM_MODEL (model), &iter);
   return info;
 }
 
-- 
2.7.1

